polus-training-splinedist-plugin/src/utils.py

```diff
@@ -0,0 +1,520 @@
+    logger.info("\n Starting to Load Train Data ...")
+    tenpercent_trained = np.floor(num_trained/10)
+    contoursize_max = 0 # gets updated if models have not been created for config file
+    for i in range(num_trained):
+
+        br_image = BioReader(X_trn[i])
+        br_image_shape = br_image.shape[:2]
+        im_array = br_image[:,:,0:1,0:1,0:1]
+        im_array = im_array.reshape(br_image_shape)
+        array_images_trained.append(normalize(im_array,pmin=1,pmax=99.8,axis=axis_norm))
+
+        if i == 0:
+            n_channel = br_image.shape[2]
+
+        br_label = BioReader(Y_trn[i])
+        lab_array = br_label[:,:,0:1,0:1,0:1]
+        lab_array = lab_array.reshape(br_label.shape[:2])
+        array_labels_trained.append(fill_label_holes(lab_array))
+        
+        assert im_array.shape == lab_array.shape, "{} and {} do not have matching shapes".format(base_image, base_label)
+        contoursize_max = update_countoursize_max(contoursize_max, lab_array)
+        
+        if (i%tenpercent_trained == 0) and (i!=0):
+            logger.info("Loaded ~{}% of Train Data -- contoursize_max: {}".format(np.ceil((i/num_trained)*100), contoursize_max))
+    logger.info("Done Loading Training Data")
+        
+
+    # Build the model and generate other necessary files to train the data.
+    logger.info("Max Contoursize: {}".format(contoursize_max))
+
+    n_params = M*2
+    grid = (2,2)
+    conf = Config2D (
+    n_params            = n_params,
+    grid                = grid,
+    n_channel_in        = n_channel,
+    contoursize_max     = contoursize_max,
+    train_epochs        = epochs,
+    use_gpu             = gpu
+    )
+    
+    del X_val
+    del Y_val
+    del X_trn
+    del Y_trn
+
+    logger.info("\n Generating phi and grids ... ")
+    if not os.path.exists("./phi_{}.npy".format(M)):
+        phi_generator(M, conf.contoursize_max, '.')
+    logger.info("Generated phi")
+    if not os.path.exists("./grid_{}.npy".format(M)):
+        grid_generator(M, conf.train_patch_size, conf.grid, '.')
+    logger.info("Generated grid")
+
+    model = SplineDist2D(conf, name=model_dir_name, basedir=output_directory)
+    del conf
+
+    # After creating or loading model, train it
+    # model.config.train_tensorboard = False
+    # model.config.use_gpu = gpu
+    logger.info("\n Parameters in Config File ...")
+    for ky,val in model.config.__dict__.items():
+        logger.info("{}: {}".format(ky, val))
+
+
+    model.train(array_images_trained,array_labels_trained, 
+                validation_data=(array_images_tested, array_labels_tested), 
+                augmenter=augmenter, epochs=epochs)
+    logger.info("\n Done Training Model ...")
+
+
+def test_nn(image_dir_test : str,
+            label_dir_test : str,
+            model_basedir : str,
+            output_directory : str,
+            gpu : bool,
+            imagepattern : str):
+    """ This function tests the trained neural network from the train_nn 
```

Comment:
As stated before, I'm not sure what the purpose of having a test function. I think you should do one of the following:

1. Remove this - I don't know what the outputs of this would look like, or how it would differ from the inference plugin. If you're training, then it's clear that the output would be a model. When you put this in test mode, the output doesn't seem like it should be a model, meaning it should be something else. If the the output is going to be a different format, this almost sounds like it should be a different plugin.
2. Provide a better explanation of this option in the README - If you have a good reason to keep this option, provide a good explanation of what the test mode is doing.
3. Turn this into a test - If you want to keep this around for testing outside of WIPP, turn it into a unit test.